thoth/storages/graph/postgres.py

```diff
@@ -3775,97 +3775,6 @@ def create_python_cve_record(
-            result = (
-                session.query(PythonPackageVersion)
-                .join(PythonPackageIndex)
-                .join(PythonRequirementsLock)
-                .join(AdviserRun, AdviserRun.user_software_stack_id == PythonRequirementsLock.python_software_stack_id)
-                .order_by(AdviserRun.origin)
-                .order_by(AdviserRun.datetime.desc())
-                .distinct(AdviserRun.origin)
-                .filter(PythonPackageIndex.url == index_url)
-                .filter(PythonPackageVersion.package_name == package_name)
-                .with_entities(AdviserRun.origin)
-            ).all()
-            result = [x[0] for x in result]
-            return result
-
-    def get_all_repositories_using_package_version(
-        self,
-        package_name: str,
-        package_version: str,
-        index_url: str,
-    ) -> List[str]:
-        """Given package version info, get unique urls of repos which have been advised with this package version."""
-        with self._session_scope() as session:
-            result = (
-                session.query(PythonPackageVersion)
-                .join(PythonPackageIndex)
-                .join(PythonRequirementsLock)
-                .join(AdviserRun, AdviserRun.user_software_stack_id == PythonRequirementsLock.python_software_stack_id)
-                .order_by(AdviserRun.origin)
-                .order_by(AdviserRun.datetime.desc())
-                .distinct(AdviserRun.origin)
-                .filter(PythonPackageIndex.url == index_url)
-                .filter(PythonPackageVersion.package_name == package_name)
-                .filter(PythonPackageVersion.package_version == package_version)
-                .with_entities(AdviserRun.origin)
-            ).all()
-            result = [x[0] for x in result]
-            return result
-
-    def remove_python_package_hash(package_name: str, package_version: str, index_url: str, sha256_hash: str):
-        """Remove hash associated with python package in the graph."""
-        with self._session_scope() as session:
-            # We need to remove rows from both HasArtifact and PythonArtifact
-            subq = (
-                session.query(PythonPackageVersionEntity)
-                .filter(PythonPackageVersionEntity.package_name == package_name)
-                .filter(PythonPackageVersionEntity.package_version == package_version)
-                .join(PythonPackageIndex)
-                .filter(PythonPackageIndex.url == index_url)
-                .join(HasArtifact)
-                .join(PythonArtifact)
-                .filter(PythonArtifact.artifact_hash_sha256 == sha256_hash)
-                .with_entities(PythonArtifact.id)
-            )
-            # Can a hash be present on more than one python_version_entity?
-            (
-                session.query(PythonArtifact)
-                .filter(PythonArtifact.id.in_(subq))
-                .delete(synchronize_session='fetch')
```

Comment:
We should be careful with deletions. We have 'on delete cascade' in our database that, in the worst case, can cause deleting the whole database when deleting a single row. Might be good to consider "active" flag. In that case we would be able to tell people "yes, we saw this hash before, but the artifact with that hash was deleted" that can be valuable info.